src/Microsoft.AspNet.Mvc.DataAnnotations/ValidationAttributeAdapterProvider.cs

```diff
@@ -0,0 +1,160 @@
+
+using System.ComponentModel.DataAnnotations;
+using Microsoft.AspNet.Mvc.ModelBinding.Validation;
+using Microsoft.Extensions.Localization;
+
+namespace Microsoft.AspNet.Mvc.DataAnnotations
+{
+    /// <summary>
+    /// Creates a <see cref="IAttributeAdapter"/> for the given attribute.
+    /// </summary>
+    public class ValidationAttributeAdapterProvider
+    {
+        /// <summary>
+        /// Creates a <see cref="IAttributeAdapter"/> for the given attribute.
+        /// </summary>
+        /// <param name="attribute">The attribute to create an adapter for.</param>
+        /// <param name="stringLocalizer">The localizer to provide to the adapter.</param>
+        /// <returns>An <see cref="IAttributeAdapter"/> for the given attribute.</returns>
+        public static IAttributeAdapter GetAttributeAdapter(ValidationAttribute attribute, IStringLocalizer stringLocalizer)
+        {
+            IAttributeAdapter adapter;
+
+            var type = attribute.GetType();
+
+            if (type == typeof(RegularExpressionAttribute))
+            {
+                adapter = GetRegularExpressionAdapter(attribute, stringLocalizer);
+            }
+            else if (type == typeof(MaxLengthAttribute))
+            {
+                adapter = GetMaxLengthAdapter(attribute, stringLocalizer);
+            }
+            else if (type == typeof(RequiredAttribute))
+            {
+                adapter = GetRequiredAdapter(attribute, stringLocalizer);
+            }
+            else if (type == typeof(CompareAttribute))
+            {
+                adapter = GetCompareAdapter(attribute, stringLocalizer);
+            }
+            else if (type == typeof(MinLengthAttribute))
+            {
+                adapter = GetMinLengthAdapter(attribute, stringLocalizer);
+            }
+            else if (type == typeof(CreditCardAttribute))
+            {
+                adapter = GetCreditCardAdapter(attribute, stringLocalizer);
+            }
+            else if (type == typeof(StringLengthAttribute))
+            {
+                adapter = GetStringLengthAdapter(attribute, stringLocalizer);
+            }
+            else if (type == typeof(RangeAttribute))
+            {
+                adapter = GetRangeAdapter(attribute, stringLocalizer);
+            }
+            else if (type == typeof(EmailAddressAttribute))
+            {
+                adapter = GetEmailAddressAdapter(attribute, stringLocalizer);
+            }
+            else if (type == typeof(PhoneAttribute))
+            {
+                adapter = GetPhoneAdapter(attribute, stringLocalizer);
+            }
+            else if (type == typeof(UrlAttribute))
+            {
+                adapter = GetUrlAdapter(attribute, stringLocalizer);
+            }
+            else
+            {
+                adapter = null;
+            }
+
+            return adapter;
+        }
+
+        private static RegularExpressionAttributeAdapter GetRegularExpressionAdapter(
```

Comment:
Hmm, thought these methods would do more. Could even go as far as to inline each of these methods. The above would become:
```C#
if (type == typeof(RegularExpressionAttribute))
{
	return new RegularExpressionAttributeAdapter(
		(RegularExpressionAttribute)attribute, 
		stringLocalizer)
}
```

Don't feel super strong but would be more terse.